pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -179,7 +181,11 @@ func createKnativeService(componentMeta metav1.ObjectMeta,
 	}
 	//Call setDefaults on desired knative service here to avoid diffs generated because knative defaulter webhook is
 	//called when creating or updating the knative service
-	service.SetDefaults(context.TODO())
+	// Before calling setDefaults load the knative configmap into the context to avoid overriding the knative configmap values.
+	ctx := context.TODO()
+	knConfigStore := knservingconfig.NewStore(knlogging.FromContext(ctx).Named(constants.KnativeReconcilerName))
+	ctx = knservingconfig.ToContext(ctx, knConfigStore.Load())
+	service.SetDefaults(ctx)
```

Comment:
I think probably not good idea to add too much Knative dependency here, calling dry-run could be a better option to get the expected object.

https://github.com/kserve/kserve/blob/1cc7e6c94eb848ba934eb75129ffba44db1fe2ed/pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go#L103